pkg/client/fake/client_test.go

```diff
@@ -2068,6 +2070,70 @@ var _ = Describe("Fake client", func() {
 		err := cl.Get(context.Background(), client.ObjectKey{Name: "foo"}, obj)
 		Expect(apierrors.IsNotFound(err)).To(BeTrue())
 	})
+
+	It("disallows scale subresources on unsupported built-in types", func() {
+		scheme := runtime.NewScheme()
+		Expect(corev1.AddToScheme(scheme)).To(Succeed())
+		Expect(apiextensions.AddToScheme(scheme)).To(Succeed())
+
+		obj := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			},
+		}
+		cl := NewClientBuilder().WithScheme(scheme).WithObjects(obj).Build()
+
+		scale := &autoscalingv1.Scale{Spec: autoscalingv1.ScaleSpec{Replicas: 2}}
+		Expect(cl.SubResource(subResourceScale).Get(context.Background(), obj, scale)).ToNot(Succeed())
+		Expect(cl.SubResource(subResourceScale).Update(context.Background(), obj, client.WithSubResourceBody(scale))).ToNot(Succeed())
+	})
+
+	It("should be able to Get scale subresources", func() {
+		obj := &appsv1.Deployment{
```

Comment:
It would be good to test this for all object kinds where we support this, maybe you can loop over a list of deployment/replicaset/statefulset/daemonset for that?